libpoolprotocols/stratum/EthStratumClient.cpp

```diff
@@ -752,37 +755,39 @@ void EthStratumClient::processReponse(Json::Value& responseObject)
 			
 			}
 			else {
-
 				cnote << "Authorized worker " + m_conn.User();
-
 			}
 			
 			break;
 
 		case 4:
 
-			// Response to solution submission mining.submit (4)
+			// Response to solution submission mining.submit  (https://en.bitcoin.it/wiki/Stratum_mining_protocol#mining.submit)
+			// Result should be boolean, some pools also throw an error, so _isSuccess can be false
+			submitSuccessful = (!_isSuccess) ? false : ((!jResult.empty() && jResult.isBool()) ? jResult.asBool() : false);
 			{
 				m_responsetimer.cancel();
 				m_response_pending = false;
-				if (_isSuccess) {
+				if (submitSuccessful) {
 					if (m_onSolutionAccepted) {
 						m_onSolutionAccepted(m_stale);
 					}
 				}
 				else {
-					cwarn << "Error :" + _errReason;
+					if (!_errReason.empty()) {
+						cwarn << "Solution rejected with the following error: " + _errReason;
```

Comment:
This would create a "double" rejected line as PoolManager does it also.

Reply:
So your suggestion would be keeping "Error: " as text?